### PR TITLE
Added hardening role

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

File: roles/hardening/files/limits.conf [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# /etc/security/limits.conf
```
Comment (carinadigital) [inline roles/hardening/files/limits.conf:1]: Is this a copy pasta ?
is this default file or did we add this line
```
*                hard    core             0
```

Comment (brettminnie) [reply roles/hardening/files/limits.conf:1]: see https://github.com/ministryofjustice/opg-hardening-formula/blob/master/hardening/templates/limits.conf

File: roles/hardening/files/limits.conf [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#
+#Each line describes a limit for a user in the form:
```
Comment (carinadigital) [inline roles/hardening/files/limits.conf:3]: Won't changes to this file require a reboot?

File: roles/hardening/tasks/sshd.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+---
+
+- name: Update sshd config file
+  copy:
+    src: sshd_config
+    dest: /etc/ssh/sshd_config
+  become: true
+
+- name: ensure that sshd is enabled and running
```
Comment (carinadigital) [inline roles/hardening/tasks/sshd.yml:16]: You not going to force a sshd restart if the config changes?

File: roles/hardening/files/limits.conf (hunk 2 above)
Comment (Azulinho) [reply roles/hardening/files/limits.conf:3]: I think this is an ansible role to be consume by the baking of the AMI, so it will be set at boot time

**Review by Azulinho — CHANGES_REQUESTED** [review]
what @carinadigital said

File: roles/hardening/files/limits.conf (hunk 2 above)
Comment (brettminnie) [reply roles/hardening/files/limits.conf:3]: Can we get it into your ami builder then and I will remove it from my playbook

File: roles/hardening/tasks/sshd.yml (hunk 3 above)
Comment (brettminnie) [reply roles/hardening/tasks/sshd.yml:16]: Good point

File: roles/hardening/files/limits.conf (hunk 2 above)
Comment (Azulinho) [reply roles/hardening/files/limits.conf:3]: and /etc/security/limits.conf will affect new sessions ( needing a reboot to make sure it catches everything)

**Pull request merged**